### PR TITLE
Feature: Ability to Login with Email

### PR DESCRIPTION
**Pull request by gsumpster**
Implemented #674, added utility class to house `findUserbyEmail($email)`, created email regex check in `classes/admin.php`'s `authenticate($data, $post)` function.

### CODE REVIEW

**Comment by flaviocopes** [comment]
@rhukster you think we can merge this feat in 1.1 or we do it in a point release after it?

**Comment by rhukster** [comment]
I think we can merge if all works as expected. Not tested myself.

Cheers!

--
Andy Miller
Sent with Airmail

File: admin.php [hunk 1]
```diff
@@ -290,6 +290,9 @@ public function onPluginsInitialized()
             };
         }
 
+        // Load utility class
+        require_once __DIR__ . '/classes/utils.php';
```
Comment (flaviocopes) [inline admin.php]: Use `use Grav\Common\Utils;` at the top, or there's a problem (try installing the Form plugin, for example, and you'll see)

Comment (flaviocopes) [inline admin.php]: Sorry, misread - it's a new file. But it triggers the problem.. do we really need a separate file just for `findUserbyEmail`?

Comment (flaviocopes) [inline admin.php]: Added my change in https://github.com/getgrav/grav-plugin-admin/pull/686/commits/b33fd299fe31648878e4b6c5d0b683ee7ed7d892, what you think?

**Comment by flaviocopes** [comment]
Did a couple fixes in #686 and merged in https://github.com/getgrav/grav-plugin-admin/commit/64a88c916cb048896582ae7433293bb82728131f, thanks!

**Pull request closed without merging**